checker.cpp

```diff
@@ -2,21 +2,57 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+class Battery{
+    public:
+        float temperature,soc,chargeRate;
+    bool checkTempIsInRange(float t);
+    bool checkSocIsInRange(float s);
+    bool checkChargeRateIsInRange(float c);
+    bool checkBatteryIsOk(float t,float s ,float c);
+};
+
+class NewBattery:public Battery
+{  
+    public:
+    bool checkTempIsInRange(float temperature)
+    {
+        if(temperature < 0 || temperature > 45)
+            return false;
+        else 
+        return true;
+    }
+    bool checkSocIsInRange(float soc)
+    {
+        if(soc < 20 || soc > 80)
+            return false;
+        else
+            return true;        
+    }
+    bool checkChargeRateIsInRange(float chargeRate)
+    {
+        if(chargeRate>0.8)
+            return false;
+        else 
+            return true;
+    }
+    bool checkBatteryIsOk(float t,float s ,float c)
+    {
+        return(checkTempIsInRange(t)&&checkSocIsInRange(s)&&checkChargeRateIsInRange(c));
+    }
+};
+
+void checkBattery()
+{
+    NewBattery n;
+    assert(n.checkBatteryIsOk(0,19,1)==false);
```

Comment:
checkTempIsInRange() is defined as public. Hence asserts maybe added to extend test to test these kind of methods. Overall having test cases to test any public method is a beneficial.

Reply:
Added asserts to check temperature, soc and chargerate is in range or not.